In [1]:
library(tidyverse)

players <- read_csv("players.csv")
sessions <- read_csv("sessions.csv")

dim(players); head(players)
dim(sessions); head(sessions)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimite

[1] 196   7

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

[1] 1535    5

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

In [5]:
# variable names
names(players)
names(sessions)

[1] "experience"   "subscribe"    "hashedEmail"  "played_hours" "name"        
[6] "gender"       "Age"

[1] "hashedEmail"         "start_time"          "end_time"           
[4] "original_start_time" "original_end_time"

In [6]:
# variable types and quick preview
glimpse(players)
glimpse(sessions)

Rows: 196
Columns: 7
$ experience   <chr> "Pro", "Veteran", "Veteran", "Amateur", "Regular", "Amate…
$ subscribe    <lgl> TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, T…
$ hashedEmail  <chr> "f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8…
$ played_hours <dbl> 30.3, 3.8, 0.0, 0.7, 0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 1.6, 0…
$ name         <chr> "Morgan", "Christian", "Blake", "Flora", "Kylie", "Adrian…
$ gender       <chr> "Male", "Male", "Male", "Female", "Male", "Female", "Fema…
$ Age          <dbl> 9, 17, 17, 21, 21, 17, 19, 21, 47, 22, 23, 17, 25, 22, 17…
Rows: 1,535
Columns: 5
$ hashedEmail         <chr> "bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8a…
$ start_time          <chr> "30/06/2024 18:12", "17/06/2024 23:33", "25/07/202…
$ end_time            <chr> "30/06/2024 18:24", "17/06/2024 23:46", "25/07/202…
$ original_start_time <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+12…
$ original_end_time   <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+

In [10]:
distinct(players, experience)
distinct(players, subscribe)
distinct(players, gender)

players |>
  summarise(
    max_played_hours = max(played_hours, na.rm = TRUE),
    min_played_hours = min(played_hours, na.rm = TRUE),
    max_age = max(Age, na.rm = TRUE),
    min_age = min(Age, na.rm = TRUE)
  )

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

max_played_hours min_played_hours max_age min_age
1 223.1            0                58      9

In [11]:
players %>%
  summarise(across(where(is.numeric), ~ round(mean(.x, na.rm = TRUE), 2)))

sessions %>%
  summarise(across(where(is.numeric), ~ round(mean(.x, na.rm = TRUE), 2)))

played_hours Age  
1 5.85         21.14

original_start_time original_end_time
1 1.719201e+12        1.719196e+12

# This project uses two datasets: players.csv and sessions.csv.


## 1. Number of observations & variables
	•	players.csv: 196 rows, 7 columns
	•	sessions.csv: 1535 rows, 5 columns

## 2. Variable names

players.csv: experience, subscribe, hashedEmail, played_hours, name, gender, Age
sessions.csv: hashedEmail, start_time, end_time, original_start_time, original_end_time

## 3. Variable types & meanings
	•	experience (chr): player skill level
	•	subscribe (lgl): subscribed or not
	•	hashedEmail (chr): anonymized ID
	•	played_hours (dbl): total hours played
	•	name (chr): chosen player name
	•	gender (chr): gender identity
	•	Age (dbl): player age
	•	start_time / end_time (chr): gameplay session time
	•	original_start_time / original_end_time (dbl): numeric timestamps

## 4. Unique categories
	•	experience: Pro, Veteran, Amateur, Regular, Beginner
	•	subscribe: TRUE, FALSE
	•	gender: Male, Female, Non-binary, Prefer not to say, Agender, Two-Spirited, Other

## 5. Summary statistics (rounded)

players.csv
	•	Mean played_hours = 5.85
	•	Played_hours range = 0 – 223.1
	•	Age range = 9 – 58 (mean ≈ 21.14)

sessions.csv
	•	Timestamps need conversion (large numeric values)

## 6. Potential Issues
	•	start_time / end_time are stored as strings
	•	timestamp columns need conversion
	•	played_hours has many zeros
	•	gender categories are many
	•	datasets must be merged via hashedEmail

